internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -206,26 +207,63 @@ func (r *BundleReconciler) ensureUnpackPod(ctx context.Context, bundle *rukpakv1
 			{Name: "util", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
+			{Name: "manifests", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
 		}
 		pod.Spec.RestartPolicy = corev1.RestartPolicyNever
-		if len(pod.Spec.InitContainers) != 1 {
-			pod.Spec.InitContainers = make([]corev1.Container, 1)
+		if bundle.Spec.Source.Type == "git" {
+			// Note: use host networking (hack) so we can resolve https://github.com references in the
+			// Bundle's spec.Source.Git.Repository. Maybe we should revisit having the user inline https://github.com?
+			pod.Spec.HostNetwork = true
+		}
+
+		targetInitContainers := 1
+		if bundle.Spec.Source.Type == "git" {
+			targetInitContainers = 2
 		}
 		pod.Spec.InitContainers[0].Name = "install-unpacker"
+		if len(pod.Spec.InitContainers) != targetInitContainers {
+			pod.Spec.InitContainers = make([]corev1.Container, targetInitContainers)
+		}
 		pod.Spec.InitContainers[0].Image = r.UnpackImage
 		pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
 		pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
 		pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
 
+		// Note: initContainer so we can ensure the repository has been cloned
+		// at the bundle.Spec.Source.Git.Ref before we unpack the Bundle contents
+		// that are stored in the repository.
+		if bundle.Spec.Source.Type == "git" {
+			pod.Spec.InitContainers[1].Name = "clone-repository"
+			pod.Spec.InitContainers[1].Image = "bitnami/git:latest"
+			pod.Spec.InitContainers[1].ImagePullPolicy = corev1.PullIfNotPresent
+			checkoutCmd := git.NewCheckoutCmd(*bundle.Spec.Source.Git)
+			c, err := checkoutCmd.String()
+			if err != nil {
+				return err
+			}
+			pod.Spec.InitContainers[1].Command = []string{"/bin/bash", "-c", c}
+			pod.Spec.InitContainers[1].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}, {Name: "manifests", MountPath: "/manifests"}}
+		}
+
 		if len(pod.Spec.Containers) != 1 {
 			pod.Spec.Containers = make([]corev1.Container, 1)
 		}
+
 		if bundle.Spec.Source.Image != nil {
 			pod.Spec.Containers[0].Name = bundleUnpackContainerName
 			pod.Spec.Containers[0].Image = bundle.Spec.Source.Image.Ref
 			pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
 			pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
 		}
+
+		if bundle.Spec.Source.Type == "git" {
+			pod.Spec.Containers[0].Name = bundleUnpackContainerName
+			pod.Spec.Containers[0].Image = "bitnami/git:latest"
+			pod.Spec.Containers[0].ImagePullPolicy = corev1.PullAlways
+			pod.Spec.Containers[0].Command = []string{"/util/unpack", "--bundle-dir", "/manifests"}
+			pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}, {Name: "manifests", MountPath: "/manifests"}}
+			return nil
+		}
```

Comment:
Should we extract the `image` pod and `git` pod into fully separate functions with just a switch statement somewhere?

I feel like that might be easier to maintain over time even if there is some duplication between them.

Thoughts?

Reply:
Yup - agreed, this needs to be done. I was talking to dan in the #rukpak-dev about whether we should attempt a refactoring of this pod unpacking method in this PR, or punt that kind of work as a follow-up.

Reply:
It totally makes sense, I was hacking around with breaking the code out and deduplicating it in https://github.com/exdx/rukpak/commit/4a2d74e4bd8f353926d3f33881c0505be9a5f556 but I'm not sure if we need something like that or it can be a follow-up. Simply breaking out into two separate functions is easier and makes more sense.

Reply:
I broke the large function out into two smaller pieces. There is still some abstraction to be desired but I think it's good enough for now.